### ISSUE TITLE
Undocumented BREAKING CHANGES

### ISSUE THREAD

**Issue by shlomiassaf** [angular/material2#4002]
@jelbourn  Hi,

While upgrading I found some undocumented breaking changes.

  - the `config` property is no longer part of `MdDialogRef`, now one needs to inject the `MD_DIALOG_DATA` token

  - `DomProjectionHost` and `DomProjection` are no longer part of `@angular/material`, couldn't find a replacement of documentation for it.

this is it for now

**Comment by ghoullier** [comment]
@shlomiassaf you can access to the dialog config via the following property

```ts
export class DialogComponent {
  constructor(private dialog: MdDialogRef<any>) {
    const config = dialog._containerInstance.dialogConfig;
  }
}
```

**Comment by shlomiassaf** [comment]
I didn't try.

The `_containerInstance` seems like a private variable so I wouldn't want to use it and later face issues.

anyway, the `MD_DIALOG_DATA` is how to get the data now...

**Comment by ghoullier** [comment]
Yes but `MD_DIALOG_DATA` allows you to retrive dialogConfig via `DialogInjector` but dialogInjector reference seems to be not exposed.

**Comment by shlomiassaf** [comment]
I don't understand what you mean...
MD_DIALOG_DATA returns the data object supplied to the confit.

I don't think the config exposed.

**Comment by damsorian** [comment]
hi @shlomiassaf .. how we use the MD_DIALOG_DATA? can you comment the example?

**Comment by xuanshenbo** [comment]
Should this be closed by [#4126](https://github.com/angular/material2/pull/4126)?

**Comment by jelbourn** [comment]
Yes
